### PR TITLE
feature: groups all action components configurations into one object per component 

### PR DESCRIPTION
**Pull request by mohammedDehairy**
- groups all action components configurations into one object per component 
- and also did a minor refactoring in VoucherComponent and DocumentComponent do be less error prone.

### CODE REVIEW

File: AdyenActions/Components/Await/AwaitComponent.swift [hunk 1]
```diff
@@ -22,35 +22,53 @@ public final class AwaitComponent: ActionComponent, Cancellable {
     /// :nodoc:
     public let requiresModalPresentation: Bool = true
     
-    /// The Component UI style.
-    public let style: AwaitComponentStyle
+    /// The await component configurations.
+    public struct Configuration {
```
Comment (erenbesel) [inline AdyenActions/Components/Await/AwaitComponent.swift:26]: I think all of them would be better conforming to `AnyBasicComponentConfiguration` for consistency.

File: Tests/AdyenTests/Adyen Tests/Components/BACS Direct Debit/Mocks/DocumentActionViewDelegateMock.swift [hunk 2]
```diff
@@ -0,0 +1,25 @@
+//
+//  DocumentActionViewDelegateMock.swift
+//  AdyenUIKitTests
+//
+//  Created by Eren Besel on 12/16/21.
```
Comment (descorp) [inline Tests/AdyenTests/Adyen Tests/Components/BACS Direct Debit/Mocks/DocumentActionViewDelegateMock.swift:5]: Unformatted headers 🙄

**Review by erenbesel — APPROVED** [review]

File: AdyenActions/Components/Await/AwaitComponent.swift [hunk 3]
```diff
@@ -22,35 +22,53 @@ public final class AwaitComponent: ActionComponent, Cancellable {
...
+        
+        /// The component UI style.
+        public var style: AwaitComponentStyle = .init()
```
Comment (kaphacius) [inline AdyenActions/Components/Await/AwaitComponent.swift:29]: do we need to set the initial value here if it is set in the `init`?

**Review by descorp — APPROVED** [review]

**Review by kaphacius — APPROVED** [review]

File: AdyenActions/Components/QRCode/QRCodeComponent.swift [hunk 4]
```diff
@@ -197,8 +216,8 @@ extension QRCodeComponent: ActionComponentDelegate {
 /// :nodoc:
 extension QRCodeComponent: QRCodeViewDelegate {
     
-    func copyToPasteboard() {
-        UIPasteboard.general.string = action.map(\.qrCodeData)
+    func copyToPasteboard(with action: QRCodeAction) {
```
Comment (kaphacius) [inline AdyenActions/Components/QRCode/QRCodeComponent.swift:219]: this should probably have access level specifier

Comment (mohammedDehairy) [reply AdyenActions/Components/QRCode/QRCodeComponent.swift:219]: can I fix this in a new PR?

File: AdyenActions/Components/Await/AwaitComponent.swift (hunk 3 above)
Comment (mohammedDehairy) [reply AdyenActions/Components/Await/AwaitComponent.swift:29]: it doesn't hurt, does it?

File: AdyenActions/Components/QRCode/QRCodeComponent.swift (hunk 4 above)
Comment (kaphacius) [reply AdyenActions/Components/QRCode/QRCodeComponent.swift:219]: yes, i approved this one :)

File: AdyenActions/Components/Await/AwaitComponent.swift (hunk 3 above)
Comment (kaphacius) [reply AdyenActions/Components/Await/AwaitComponent.swift:29]: it's just settings a value which will immediately be overwritten in `init`. if it had no initial value - compiler would check if every version of `init` actually sets a value for this field.

Comment (mohammedDehairy) [reply AdyenActions/Components/Await/AwaitComponent.swift:29]: ok will fix it with the other one in another PR

**Pull request merged**